html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/data-url-agent-cluster.https.html

```diff
@@ -0,0 +1,45 @@
+<!doctype html>
+<meta charset=utf-8>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/wasm/jsapi/wasm-module-builder.js"></script>
+<div id=log></div>
+<script>
+const dataURLWorker = `
+  onmessage = () => self.postMessage("FAIL");
+  onmessageerror = () => self.postMessage("PASS");
+`;
+
+const dataFrameScript = `
+  onmessage = () => parent.postMessage("FAIL", "*");
+  onmessageerror = () => parent.postMessage("FAIL", "*");
+`;
+
+[
+  new SharedArrayBuffer(),
+  new WebAssembly.Module(new WasmModuleBuilder().toBuffer())
+].forEach(instance => {
+  async_test(t => {
+    const worker = new Worker(`data:,${dataURLWorker}`);
+    worker.onmessage = t.step_func_done(({ data }) => {
+      assert_equals(data, "PASS");
+    });
+    worker.postMessage(instance);
+  }, `As data URL dedicated workers form their own agent cluster, ${instance.constructor.name} cannot be messaged to them`);
+
+  promise_test(t => {
+    return new Promise(resolve => {
+      const frame = document.body.appendChild(document.createElement("iframe"));
+      t.add_cleanup(() => frame.remove());
+      frame.onload = t.step_func(() => {
+        self.addEventListener("message", t.step_func(({ data }) => {
+          assert_equals(data, "PASS");
```

Comment:
I think this can never happen since both message and messageerror will cause a "FAIL" to be posted? More reason to change the messages posted in the way I describe above.